iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,402 @@
+                return await Task.FromResult(new MethodResponse((int)StatusCode.BadRequest));
+            }
+
+            return new MethodResponse((int)StatusCode.Completed);
+        }
+
+        // The callback to handle "getMaxMinReport" command. This method will returns the max, min and average temperature from the
+        // specified time to the current time.
+        private async Task<MethodResponse> HandleMaxMinReportCommandAsync(MethodRequest request, object userContext)
+        {
+            try
+            {
+                string componentName = (string)userContext;
+                DateTime sinceInUtc = JsonConvert.DeserializeObject<DateTime>(request.DataAsJson);
+                var sinceInDateTimeOffset = new DateTimeOffset(sinceInUtc);
+
+                if (_temperatureReadingsDateTimeOffset.ContainsKey(componentName))
+                {
+                    _logger.LogDebug($"Command: Received - component=\"{componentName}\", generating max, min and avg temperature " +
+                        $"report since {sinceInDateTimeOffset.LocalDateTime}.");
+
+                    Dictionary<DateTimeOffset, double> allReadings = _temperatureReadingsDateTimeOffset[componentName];
+                    Dictionary<DateTimeOffset, double> filteredReadings = allReadings.Where(i => i.Key > sinceInDateTimeOffset)
+                        .ToDictionary(i => i.Key, i => i.Value);
+
+                    if (filteredReadings != null && filteredReadings.Any())
+                    {
+                        var report = new
+                        {
+                            maxTemp = filteredReadings.Values.Max<double>(),
+                            minTemp = filteredReadings.Values.Min<double>(),
+                            avgTemp = filteredReadings.Values.Average(),
+                            startTime = filteredReadings.Keys.Min(),
+                            endTime = filteredReadings.Keys.Max(),
+                        };
+
+                        _logger.LogDebug($"Command: component=\"{componentName}\", MaxMinReport since {sinceInDateTimeOffset.LocalDateTime}:" +
+                            $" maxTemp={report.maxTemp}, minTemp={report.minTemp}, avgTemp={report.avgTemp}, startTime={report.startTime.LocalDateTime}, " +
+                            $"endTime={report.endTime.LocalDateTime}");
+
+                        byte[] responsePayload = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(report));
+                        return await Task.FromResult(new MethodResponse(responsePayload, (int)StatusCode.Completed));
+                    }
+
+                    _logger.LogDebug($"Command: component=\"{componentName}\", no relevant readings found since {sinceInDateTimeOffset.LocalDateTime}, " +
+                        $"cannot generate any report.");
+                    return await Task.FromResult(new MethodResponse((int)StatusCode.NotFound));
```

Comment:
I think here you can just do `return new MethodResponse((int)StatusCode.NotFound)`